tests/integration/ranker_training/offline/test_offline_training.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pickle
+
+import pytest
+
+from jina import Document, Flow
+from jina.types.sets import DocumentSet
+
+'''
+User -> Train request -> RankTrainer Train -> RankTrainer Dump Weights/Parameters -> To be loaded in Ranker
+price is random and size is related to relevance to see after training, relevance is sorted based on size
+2 flows one train and one rank
+'''
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+@pytest.fixture
+def documents_to_train():
+    """The relevance is correlated to the size feature."""
+    queries = []
+    for q in range(1, 100):
+        query = Document()
+        for i in range(1, 50):
+            match = Document()
+            # large size higher relevance
+            match.tags['price'] = 1
+            match.tags['size'] = i * 2
+            match.tags['relevance'] = i
+            query.matches.add(match)
+        queries.append(query)
+    return DocumentSet(queries)
+
+
+@pytest.fixture
+def doc_to_query():
+    doc = Document()
+    for i in range(1, 5):
+        match = Document()
+        match.tags['price'] = 1
+        match.tags['size'] = i * 2
+        doc.matches.add(match)
+    return doc
+
+
+def test_train_offline(documents_to_train, doc_to_query):
+    def validate_ranking_by_price(req):
+        pred = set()
+        for match in req.docs[0].matches:
+            pred.add(match.score.value)
+        assert len(pred) == 1  # since price tag never changes, all scores are the same.
```

Comment:
do not understand this? What does it prove?

Reply:
@JoanFM we're training a model based on 2 features, `size` and `price`, in this case, the model was trained on `price` feature, and our `doc_to_query` has a fixed price value. So the predicted results are always the same

Reply:
ok, make a long documentation about this